app/controllers/publishers/omniauth_callbacks_controller.rb

```diff
@@ -1,11 +1,28 @@
 module Publishers
   class OmniauthCallbacksController < Devise::OmniauthCallbacksController
     def google_oauth2
-      publisher = current_publisher
       oauth_response = request.env['omniauth.auth']
 
       refresh_eyeshade = false
 
+      existing_publisher = Publisher.where(auth_provider: oauth_response.provider, auth_user_id: oauth_response.uid).
+          where.not(youtube_channel_id: nil).first
+
+      # The presence of a current_publisher may indicate we're attempting to attach a channel to an email verified
+      # publisher OR it's an attempt to log in using oauth.
+      #
+      # If there's an existing publisher with the same oath credentials AND the same email this is an attempt to
+      # register the same channel to effectively the same email. Instead of showing the already registered message
+      # the current_publisher can be deleted and then log them in as the existing_publisher
+      if existing_publisher && current_publisher && current_publisher.youtube_channel_id.nil? &&
+          existing_publisher.email == current_publisher.email &&
+          existing_publisher.id != current_publisher.id
+        current_publisher.destroy
```

Comment:
This should be a very uncommon case, but I'd like to track it in case we made any oops. Could we send an event to Sentry here?